backend/editor/api.py

```diff
@@ -35,4 +57,214 @@ async def pong(response: Response):
+
+@app.get("/nodesfull")
+async def findAllNodes(response: Response):
+    """
+    Get all nodes within taxonomy
+    """
+    query = """
+        MATCH (n) RETURN n
+    """
+    result = session.run(query)
+    allNodes = [record for record in result]
+    return allNodes
+
+@app.get("/entry/{entry}")
+async def findOneEntry(response: Response, entry: str):
+    """
+    Get entry corresponding to id within taxonomy
+    """
+    query = """
+        MATCH (n:ENTRY) WHERE n.id = $id 
+        RETURN n
+    """
+    result = session.run(query, {"id": entry})
+    oneEntry = [record for record in result]
+    return oneEntry
+
+@app.get("/entriesfull")
+async def findAllEntries(response: Response):
+    """
+    Get all entries within taxonomy
+    """
+    query = """
+        MATCH (n:ENTRY) RETURN n
+    """
+    result = session.run(query)
+    allEntries = [record for record in result]
+    return allEntries
+
+@app.get("/synonym/{synonym}")
+async def findOneSynonym(response: Response, synonym: str):
+    """
+    Get synonym corresponding to id within taxonomy
+    """
+    query = """
+        MATCH (n:SYNONYMS) WHERE n.id = $id 
+        RETURN n
+    """
+    result = session.run(query, {"id": synonym})
+    oneSynonym = [record for record in result]
+    return oneSynonym
+
+@app.get("/synonymsfull")
+async def findAllSynonyms(response: Response):
+    """
+    Get all synonyms within taxonomy
+    """
+    query = """
+        MATCH (n:SYNONYMS) RETURN n
+    """
+    result = session.run(query)
+    allSyononyms = [record for record in result]
+    return allSyononyms
+
+@app.get("/stopword/{stopword}")
+async def findOneStopword(response: Response, stopword: str):
+    """
+    Get stopword corresponding to id within taxonomy
+    """
+    query = """
+        MATCH (n:STOPWORDS) WHERE n.id = $id 
+        RETURN n
+    """
+    result = session.run(query, {"id": stopword})
+    oneStopword = [record for record in result]
+    return oneStopword
+
+@app.get("/stopwordsfull")
+async def findAllStopwords(response: Response):
+    """
+    Get all stopwords within taxonomy
+    """
+    query = """
+        MATCH (n:STOPWORDS) RETURN n
+    """
+    result = session.run(query)
+    allStopwords = [record for record in result]
+    return allStopwords
+
+
+@app.get("/header")
+async def findHeader(response: Response):
+    """
+    Get __header__ within taxonomy
+    """
+    query = """
+        MATCH (n:TEXT) WHERE n.id = "__header__"
+        RETURN n
+    """
+    result = session.run(query)
+    header = [record for record in result]
+    return header
+
+@app.get("/footer")
+async def findFooter(response: Response):
+    """
+    Get __footer__ within taxonomy
+    """
+    query = """
+        MATCH (n:TEXT) WHERE n.id = "__footer__"
+        RETURN n
+    """
+    result = session.run(query)
+    footer = [record for record in result]
+    return footer
+
+# Put methods 
+
+@app.put("/edit/entry/{entry}")

```

Comment:
You did a "PUT" while you method reminds more of a "POST" in the sens that it permits partial updates (normally POST enables partial updates, but not PUT). see https://en.wikipedia.org/wiki/Representational_state_transfer#Semantics_of_HTTP_methods

Reply:
Yep, I wasn't sure of the method for partial updates. I'll change it.